commit b38351a6882ca01a1b57dfd2bea0b2201b0c0d86
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2019-05-01T00:23:01+00:00 (5 months ago)
    
    joyent/node-zkstream#43 watches being lost in reattach, due to lack of `watchersDisconnected()`
    Reviewed by: Cody Peter Mello <cody.mello@joyent.com>
    Approved by: Cody Peter Mello <cody.mello@joyent.com>

diff --git a/lib/zk-session.js b/lib/zk-session.js
index 277cbec..774a7b0 100644
--- a/lib/zk-session.js
+++ b/lib/zk-session.js
@@ -280,6 +280,7 @@ ZKSession.prototype.state_reattaching = function (S) {
 		self.zs_sessionId = sid;
 		self.zs_passwd = pkt.passwd;
 		self.resetExpiryTimer();
+		self.watchersDisconnected();
 		S.gotoState('attached');
 	});
 
diff --git a/package.json b/package.json
index 1e4ef23..5ee4dfc 100644
--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "zkstream",
-  "version": "0.11.3",
+  "version": "0.11.4",
   "description": "",
   "repository": {
     "type": "git",
